common/src/main/java/com/microsoft/identity/common/adal/internal/net/DefaultConnectionService.java

```diff
@@ -107,10 +109,12 @@ public void onLost(@NonNull Network network) {
      */
     @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
     public void unregisterNetworkCallback() {
-        final ConnectivityManager connectivityManager = (ConnectivityManager) mConnectionContext
-                .getSystemService(Context.CONNECTIVITY_SERVICE);
+        if (networkCallback != null) {
```

Comment:
Is there some particular reason to lazy-initialize this?  If there is, we need to do it right, meaning we’ll have to synchronize it somehow.  It might be simpler to do in an initializer, though.

Reply:
The only reason is to make sure that `registerNetworkCallback()` does not re-register the callback when invoked multiple times.

Reply:
Right, but if you registered your callback on static initialization, if you can, then you don't need to do this at all.

Reply:
I was looking into the existing code and seems like this function is never called anywhere.